dace/transformation/subgraph/temporal_vectorization.py

```diff
@@ -0,0 +1,166 @@
+import dace
+from dace import data, dtypes, nodes, properties
+from dace.libraries.standard import Gearbox
+from dace.memlet import Memlet
+from dace.sdfg.graph import SubgraphView
+from dace.sdfg.sdfg import SDFG
+from dace.sdfg.state import SDFGState
+from dace.transformation import transformation
+from dace.transformation.subgraph import helpers
+
+
+@properties.make_properties
+class TemporalVectorization(transformation.SubgraphTransformation):
+    '''
+    This transformation applies the multi-pumping optimization to a subgraph targeting FPGAs, in turn packing more computations temporally rather than spatially as is done in traditional vectorization.
+
+    Currently, it can only be applied to applications targeting Xilinx FPGAs, and to subgraphs that purely communicate through streams. It can be applied in two ways:
+    1. Where the widths of the internal paths remain unchanged while the external paths are widened by the multi-pumping factor. This gives the benefit of increased throughput at the same critical-resource footprint.
+    2. Where the widths of the internal paths are divided by the multi-pumping factor, while the widths of the external paths remain unchanged. This gives the benefit of a reduced critical-resource footprint at the same throughput. 
+    '''
+    # TODO 3rd approach: where the subgraph is just clocked faster without introducing gearboxing. This could help designs where the subgraph reaches II=2, then by clocking it two times faster, it essentially behaves as II=1 from the slow clock domain.
+    factor = properties.Property(dtype=int,
+                                 default=2,
+                                 desc='The multi-pumping factor. E.g. double-pumping is a factor of 2.')
+    approach = properties.Property(
+        dtype=int,
+        default=2,
+        desc=
+        'Which approach to use. Can be 1 (increased throughput, same resources) or 2 (same throughput, reduced resourced).'
+    )
+
+    def can_be_applied(self, sdfg: SDFG, subgraph: SubgraphView) -> bool:
+        '''
+        Temporal vectorization can be applied if:
+        1. There is one outermost map in the subgraph. 
+        2. All of the non- source and sink nodes are only accessed within this subgraph. 
+        3. All of the source and sink nodes are either streams or scalars. 
+        4. If the direction is 1, then either:
```

Comment:
What do you mean by "the direction is 1"?

Reply:
Ah, I think you are referring to the two "ways" defined before. Pls make it more clear. Later on you call this approach

Reply:
Right! I missed that one. I changed from 'direction' to 'approach'.